### PR TITLE
Add AWS S3 Server Side Encryption support.

### PR DESCRIPTION
**Pull request by Aevin1387**
Hello! 

This adds an environment variable to enable S3 Server Side Encryption support. The S3 manager handles the rest after the configuration option is set.

### CODE REVIEW

**Comment by fdr** [comment]
A nice feature to have. Is there any way a test could be written confirming that configuration is passed through to the struct correctly? IIRC we've struggled with the meaningfulness of such a test in the past, but do you have any ideas?

**Comment by fdr** [comment]
Also, how does one specify particular KMS keys? I haven't used that feature but if we're going to add configuration complexity I want to understand its extent of completeness.

**Comment by Aevin1387** [comment]
I'm sorry for the delay on this. I've finally gotten around to testing using KMS keys. To use a KMS key, a user would set `WALG_S3_SSE` to `aws:kms` and set `WALG_S3_SSE_KMS_ID` to the key ID.

**Comment by Aevin1387** [comment]
@fdr When you have a chance, can you please take a look? I've tested this with both `AES256` and `aws:kms`, and added error handling if server side encryption is set to `aws:kms` and the KMS Key Id is not set.

File: upload.go [hunk 1]
```diff
@@ -134,6 +134,20 @@ func Configure() (*TarUploader, *Prefix, error) {
...
+	if ok {
+		upload.ServerSideEncryption = serverSideEncryption
+	}
+
+	sseKmsKeyId, ok := os.LookupEnv("WALG_S3_SSE_KMS_ID")
+	if ok {
+		upload.SSEKMSKeyId = sseKmsKeyId
+	}
+
+	if serverSideEncryption == "aws:kms" && sseKmsKeyId == "" {
```
Comment (x4m) [inline upload.go]: I think we should add some validation code here.
If user have sseKmsKeyId but no serverSideEncryption WAL-G should print an error, if serverSideEncryption is unknown, WAL-G should print warning.

File: upload.go [hunk 2]
```diff
@@ -171,12 +185,22 @@ func (tu *TarUploader) upload(input *s3manager.UploadInput, path string) (err er
...
 		Bucket:       aws.String(tu.bucket),
 		Key:          aws.String(path),
 		Body:         reader,
 		StorageClass: aws.String(tu.StorageClass),
 	}
+
+	if tu.ServerSideEncryption != "" {
+		uploadInput.ServerSideEncryption = aws.String(tu.ServerSideEncryption)
+
+		if tu.ServerSideEncryption == "aws:kms" && tu.SSEKMSKeyId != "" {
```
Comment (x4m) [inline upload.go]: SSEKMSKeyId should be setted only for ServerSideEncryption, right?
I'm a bit worried about checking string literal "aws:kms" here, validation logic is duplicated now... but probably it is OK.

**Comment by x4m** [comment]
Cory, please review my changes. If everything is OK. let's merge it.

**Comment by Aevin1387** [comment]
Just tested w/ your changes, looks good to me. :shipit:

**Comment by x4m** [comment]
Thanks!

**Pull request merged**